### ISSUE TITLE
Panic after despawning camera

### ISSUE THREAD

**Issue by geckoxx** [bevyengine/bevy#4227]
bevy version: main 026563cb1e89de30af5b801c89e8fb51a83e4970

When spawning a camera and despawning it later it panics with
```
thread 'main' panicked at 'view entity should exist: QueryDoesNotMatch', [...]\crates\bevy_core_pipeline\src\main_pass_2d.rs:45:14
```

```rust
use bevy::prelude::*;

fn main() {
    App::new()
        .add_plugins(DefaultPlugins)
        .add_startup_system(setup)
        .add_system(remove_camera)
        .run();
}

fn setup(mut commands: Commands) {
    commands.spawn_bundle(OrthographicCameraBundle::new_2d());
}

fn remove_camera(
    mut commands: Commands,
    query: Query<Entity, With<Camera>>,
    keyboard_input: Res<Input<KeyCode>>,
) {
    if keyboard_input.just_pressed(KeyCode::Space) {
        for entity in query.iter() {
            commands.entity(entity).despawn();
        }
    }
}
```

**Comment by kirusfg** [comment]
After a first incorrect impression that it's behaving correctly, I quickly realized that Bevy should be able to run without a camera. Then I remembered that #3635 was merged not so long ago, and tested your example on that commit and a commit just prior to that. The conclusion is that #3635 introduces this bug (well, "introduces" is a strong word, it's just that Bevy behaves as expected without this change).

I'll investigate further into this.
